python/cuspatial/cuspatial/core/spatial/distance.py

```diff
@@ -213,3 +221,132 @@ def pairwise_linestring_distance(offsets1, xs1, ys1, offsets2, xs2, ys2):
             )
         }
     )
+
+
+def pairwise_point_linestring_distance(
+    points: GeoSeries, linestrings: GeoSeries
+):
+    """Compute distance between pairs of (multi)points and (multi)linestrings
+
+    The distance between a (multi)point and a (multi)linestring
+    is defined as the shortest distance between every point in the
+    multipoint and every line segment in the multilinestring.
+
+    This algorithm computes distance pairwise. The ith row in the result is
+    the distance between the ith (multi)point in `points` and the ith
+    (multi)linestring in `linestrings`.
+
+    Parameters
+    ----------
+    points : GeoSeries
+        The (multi)points to compute the distance from.
+    linestrings : GeoSeries
+        The (multi)linestrings to compute the distance from.
+
+    Returns
+    -------
+    distance : cudf.Series
+
+    Notes
+    -----
+    The input `GeoSeries` must contain a single type geometry.
```

Comment:
```suggestion
    The input `GeoSeries` should contain a single type geometry.
```

I don't think there's a problem passing in a GeoSeries that has other feature types in it, they'll just be ignored. Theoretically if a GeoSeries contained an equal number of points and linestrings it could be the first and second argument to this function? I think?

Reply:
The result of a pairwise distance function should be of the same length as its input. I would rather the user explicitly extract the points and linestring from their geoseries to form a new one before passing in this function. Do we have an easy way for user to do that?

Reply:
No. In `geopandas`, the syntax commonly used is `points = geoseries[geoseries.type == "Point"]` and we need to implement some kind of the `.type` function in order to duplicate it. It can be hacked by using the `GeoColumn._meta` but we should just write the `.type` accessor. :)